[Mypy fails](https://builds.sr.ht/~whynothugo/job/1732439#task-lint-15). Pretty sure it's a false positive, but it can't infer that you're converting `dt` into the correct type:

```
> mypy .
todoman/model.py:224: error: Argument 1 to "after" of "rrulebase" has incompatible type "date"; expected "datetime"  [arg-type]
todoman/model.py:227: error: Item "None" of "datetime | None" has no attribute "date"  [union-attr]
todoman/model.py:227: error: Incompatible types in assignment (expression has type "date | Any", variable has type "datetime | None")  [assignment]
Found 3 errors in 1 file (checked 24 source files)
```

--- a/todoman/model.py
+++ b/todoman/model.py
@@ -210,17 +210,22 @@ class Todo:
     def is_recurring(self) -> bool:
         return bool(self.rrule)
 
-    def _apply_recurrence_to_dt(self, dt: date | None) -> datetime | None:
+    def _apply_recurrence_to_dt(self, dt: date | None) -> date | datetime | None:
         if not dt:
             return None
 
         assert self.rrule is not None, "applying recurrence to todo without rrule"
         recurrence = rrulestr(self.rrule, dtstart=dt)
 
-        if isinstance(dt, date) and not isinstance(dt, datetime):
+        if dateonly := not isinstance(dt, datetime):
             dt = datetime.combine(dt, time.min)
 
-        return recurrence.after(dt)
+        nxt = recurrence.after(dt)
+
+        if dateonly and nxt is not None:
+            return nxt.date()
+
+        return nxt
 
     def _create_next_instance(self) -> Todo:
         copy = self.clone()

